benchmark/kvbench/test/sequential_custom_traffic_perftest.py

```
@@ -71,37 +114,452 @@ def __init__(
+            first_storage_rank = min(storage_ranks) if storage_ranks else None
+            if self.my_rank != first_storage_rank:
+                continue
+
+            # Use single handle directly (more efficient than _run_tp for single handle)
+            handle = handles[0] if handles else None
+            if not handle:
+                continue
+
+            iter_latencies = []
+            for _ in range(self.n_isolation_iters):
+                iter_latencies.append(self._run_handle(handle))
```

Ruff (0.15.14)

[warning] Use a list comprehension to create a transformed list

(PERF401)